src/web/client/WebExtensionContext.ts

```diff
@@ -645,6 +657,47 @@ class WebExtensionContext implements IWebExtensionContext {
     public getVscodeWorkspaceState(key: string): IEntityInfo | undefined {
         return this._vscodeWorkspaceState.get(key);
     }
+
+    // Graph client
+    public async graphClientAuthentication(firstTimeAuth = false) {
+        const accessToken = await graphClientAuthentication(firstTimeAuth);
+        this.setGraphToken(accessToken);
+    }
+
+    public setGraphToken(graphToken: string) {
+        this._graphToken = graphToken;
+    }
+
+    private async getUserEmail(userId: string) {
+        try {
+            const response = await requestGraphClient(this.graphToken, Constants.GraphClientService.MAIL, userId);
+            return await response.mail;
```

Comment:
suggestion is to log specific errors here and not in the "requestGraphCLient"function - that way you can define the "entityName" differently for mail and profile pic call

Reply:
![image](https://github.com/microsoft/powerplatform-vscode/assets/56073559/6dbcc829-11f9-48e0-b0c5-f0f5e36ace94)
logging service name now in requestGraphClient, does this address this?